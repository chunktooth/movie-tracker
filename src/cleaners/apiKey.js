export const apiKey = '480eafc74b027454d89765098061cb64';

// jared - 480eafc74b027454d89765098061cb64
// Matt - 4eabf2c86624eba27b3bed0627180793
// om - e7f886548071b04cf8febc293d108511