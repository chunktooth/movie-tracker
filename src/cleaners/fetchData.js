import { apiKey } from './apiKeys';

const rootUrl = 'https://api.themoviedb.org/3';
const nowPlaying = `/movie/now_playing?page=1&language=en-US&api_key=${apiKey}&language=en-US`;

export const getNowPlaying = async () => {

  try {
    const response = await fetch(`${rootUrl}${nowPlaying}`);
    const nowPlayingData = await response.json();

    return nowPlayingData;
  } catch (error) {
    throw new Error ('its bad');
  }
};


export const addToFavorites = async (movie) => {

  try {
    const response = await fetch('/api/users/favorites/new', {
      method: 'POST',
      body: JSON.stringify(movie),
      headers: {
        'content-type': 'application/json'
      }
    });
    const favoriteData = await response.json();

    return favoriteData;
  } catch (error) {
    console.log(error);
  }
}

export const removeFromFavorites = async (userId, movieId) => {

  try {
    const response = await fetch(`/api/users/${userId}/favorites/${movieId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      }
    });
    const removedData = await response.json();
    
    return removedData;
  } catch (error) {
    console.log(error);
  }
}