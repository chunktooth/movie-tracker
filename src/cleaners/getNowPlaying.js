import { apiKey } from './apiKeys';

export const getNowPlaying = async () => {

  try {
    const response = await fetch(`https://api.themoviedb.org/3/movie/now_playing?page=1&language=en-US&api_key=${apiKey}&language=en-US`);
    const nowPlayingData = await response.json();
    const cleanedMovies = cleanMovies(nowPlayingData)
    return cleanedMovies
  } catch (error) {
    throw new Error ('Cannot fetch');
  }
};

const cleanMovies = (movieData) => {
  return movieData.results.map(movie => ({
    movie_id: movie.id, 
    title: movie.title, 
    poster_path: movie.poster_path, 
    release_date: movie.release_date, 
    vote_average: movie.vote_average, 
    overview: movie.overview  
  }));
}