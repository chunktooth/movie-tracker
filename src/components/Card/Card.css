.tile .selected{
  background-image: url('../../images/checked.svg');
  background-repeat: no-repeat; 
  background-color: transparent;
  background-size: cover;  
  width: 35px;
  height: 35px;
}