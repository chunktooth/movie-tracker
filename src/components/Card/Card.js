import React from 'react';
import './Card.css';
import { addToFavorites, removeFromFavorites } from '../../cleaners/fetchData';
import { addFavorite, removeFavorite } from '../../actions/';
import { connect } from 'react-redux';

export const Card = ({ movieInfo, user, addFavorite, favorites, removeFavorite, selected }) => {

  const userMovie = {...movieInfo, user_id: user.id};
  const {title, poster_path, overview, vote_average, movie_id, user_id, release_date} = userMovie;

  const addFavoritesToStore = (userMovie) => {

    const isInFavorites = favorites.filter(favorite => {
      return favorite.movie_id === movie_id;
    });

    if (!isInFavorites.length) {
      addToFavorites(userMovie);
      addFavorite(userMovie);
    } else {
      removeFavorite(movie_id)
      removeFromFavorites(user_id, movie_id);
    }
  }

  const validateUser = () => {  
    if (!user.id) {
      alert('Please log in or sign up to add favorites')
    } else {
      addFavoritesToStore(userMovie);

    }
  }

  return (
    <div className='tile'>
      <img className='tile__img'
          src={`https://image.tmdb.org/t/p/w500/${poster_path}`}
          alt='movie poster' />
      <div className='card-details'>
        <h2>{title}</h2>
        <textarea className='tile__title'>{overview}</textarea>
        <button 
          onClick={() => validateUser()}
          className={`${selected}`}>
          favorite</button>
      </div>
    </div>

  );
};

export const mapStateToProps = (state) => ({
  user: state.user,
  favorites: state.favorites
})

export const mapDispatchToProps = (dispatch) => ({
  addFavorite: (movie) => (dispatch(addFavorite(movie))),
  removeFavorite: (movie_id) => (dispatch(removeFavorite(movie_id))) 
});

export default connect(mapStateToProps, mapDispatchToProps)(Card)




