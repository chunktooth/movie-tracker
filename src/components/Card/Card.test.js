import React from 'react';
import { Card } from './Card';
import { mockCleanMovie, mockUser } from '../../cleaners/mockData';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  const mockAddFavorite = jest.fn();
  const mockRemoveFavorite = jest.fn();
  const mockFavorites = [];

  beforeEach(() => {
    wrapper = shallow(<Card 
    movieInfo={mockCleanMovie}
    favorites={mockFavorites}
    addFavorite={mockAddFavorite}
    removeFavorite={mockRemoveFavorite}
    user={mockUser}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it.skip('should invoke validateUser when clicking a button', () => {
    wrapper.find('button').simulate('click');
    wrapper.instance().validateUser = jest.fn();
    expect(wrapper.instance().validateUser).toHaveBeenCalled();
  });

    describe.skip('add favorite to store', () => {

      it('should addToFavorites when the movie is not in favorites', () => {
        console.log(wrapper.debug());
        wrapper.instance().addFavoritesToStore(mockCleanMovie);
      });
      
    });

});