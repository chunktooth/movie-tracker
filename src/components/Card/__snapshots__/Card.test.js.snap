// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addFavorite={[Function]}
    favorites={Array []}
    movieInfo={
        Array [
            Object {
              "movie_id": "284054",
              "overview": "King T'Challa returns home from America to the reclusive, technologically advanced African nation of Wakanda to serve as his country's new leader. However, T'Challa soon finds that he is challenged for the throne by factions within his own country as well as without.  Using powers reserved to Wakandan kings, T'Challa assumes the Black Panther mantel to join with girlfriend Nakia, the queen-mother, his princess-kid sister,  members of the Dora Milaje (the Wakandan 'special forces'), and an American secret agent, to prevent Wakanda from being dragged into a world war.",
              "poster_path": "/uxzzxijgPIY7slzFvMotPv8wjKA.jpg",
              "release_date": "2018-02-13",
              "title": "Black Panther",
              "vote_average": "7.4",
            },
          ]
    }
    removeFavorite={[Function]}
    user={
        Object {
            "id": 4,
            "name": "mud",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="movie poster"
          className="tile__img"
          src="https://image.tmdb.org/t/p/w500/undefined"
/>,
        <div
          className="card-details"
>
          <h2 />
          <textarea
                    className="tile__title"
          />
          <button
                    className="undefined"
                    onClick={[Function]}
          >
                    favorite
          </button>
</div>,
      ],
      "className": "tile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "movie poster",
          "className": "tile__img",
          "src": "https://image.tmdb.org/t/p/w500/undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <textarea
              className="tile__title"
/>,
            <button
              className="undefined"
              onClick={[Function]}
>
              favorite
</button>,
          ],
          "className": "card-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "tile__title",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "favorite",
              "className": "undefined",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "favorite",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="movie poster"
            className="tile__img"
            src="https://image.tmdb.org/t/p/w500/undefined"
/>,
          <div
            className="card-details"
>
            <h2 />
            <textarea
                        className="tile__title"
            />
            <button
                        className="undefined"
                        onClick={[Function]}
            >
                        favorite
            </button>
</div>,
        ],
        "className": "tile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "movie poster",
            "className": "tile__img",
            "src": "https://image.tmdb.org/t/p/w500/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <textarea
                className="tile__title"
/>,
              <button
                className="undefined"
                onClick={[Function]}
>
                favorite
</button>,
            ],
            "className": "card-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "tile__title",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "favorite",
                "className": "undefined",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "favorite",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
